### PR TITLE
feat(omnect-os-initramfs): in flash mode 2 allow direct flashing

### PR DESCRIPTION
**Pull request by HarryWaschkeit**
Previously, the provided wic image was verified against the provided bmap file which requires both of them stored in memory (tmpfs).

For devices with low memory this feature allows to directly program the image into the flash device, of course accompanied by the danger to brick a device in case a corrupted wic file is passed on to the device.

Default behavior is still to first verify, to enable direct flashing the new bitbake variable OMNECT_FLASH_MODE_2_DIRECT_FLASHING needs to be set to "1".

### CODE REVIEW

File: recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2 [hunk 1]
```diff
@@ -56,14 +58,18 @@ run_bmap_tool() {
...
+	run_cmd_stdout bmaptool copy --bmap /home/omnect/wic.bmap \
+		       ${wic_file} /home/omnect/wic || return 1
+	wic_file=/home/omnect/wic
+	msg "successfully verified input wic.xz"
+    fi
     vfat_dd_handling "/dev/omnect/boot"
+    msg "start flashing ..."
     run_cmd_stdout bmaptool copy --bmap /home/omnect/wic.bmap \
-        /home/omnect/wic /dev/omnect/rootblk || return 1
+		       $wic_file /dev/omnect/rootblk || return 1
```
Comment (mlilien) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2:72]: ${wic_file} vss $wic_file

File: recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2 [hunk 2]
```diff
@@ -44,7 +46,7 @@ run_bmap_tool() {
         fi
     done </tmp/all_mounts
 
-    run_cmd mkfifo /home/omnect/wic.xz || return 1
+    run_cmd mkfifo ${wic_file} || return 1
     run_cmd chown omnect:omnect /home/omnect/wic.xz || return 1
```
Comment (mlilien) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2:50]: you introduced ${wic_file}, but there are still usages of `/home/omnect/wic.xz` and `/home/omnect/wic` and `/home/omnect.wic.bmap`.
so why use it in a var only in some locations?

Comment (HarryWaschkeit) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2]: wic_file was intended for image file only, so no use for bmap file.
and /home/omnect/wic `/home/omnect/wic` is used together with wic_file during verification, and wic_file adapted accordingly afterward.

File: recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2 [hunk 3]
```diff
@@ -33,7 +34,7 @@ flash_mode_2_enabled() {
 run_bmap_tool() {
     local umount_dev=""
     local bmap_ret=""
-
+
```
Comment (mlilien) [inline recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2:37]: please check your editor settings to trim whitespaces at eol

Comment (HarryWaschkeit) [reply recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2]: done

**Review by mlilien — APPROVED** [review]

**Pull request merged**